apps/l10nstats/views.py

```diff
@@ -290,29 +329,45 @@ def increment(self):
         return str(self.count)
 
 
-def compare(request):
+class CompareView(TemplateView):
     """HTML pretty-fied output of compare-locales.
     """
-    try:
-        run = get_object_or_404(Run, id=request.GET.get('run'))
-    except ValueError:
-        return HttpResponseBadRequest('Invalid ID')
-    # try disk first, then ES
-    jsondata = ''
-    doc = None
-    for step in Step.objects.filter(name__startswith='moz_inspectlocales',
-                                    build__run=run):
-        for log in step.logs.all():
-            try:
-                for chunk in generateLog(run.build.builder.master.name,
-                                         log.filename,
-                                         channels=(Log.JSON,)):
-                    jsondata += chunk['data']
-            except NoLogFile:
-                pass
-    if jsondata:
-        doc = json.loads(jsondata)
-    elif hasattr(settings, 'ES_COMPARE_HOST'):
+    template_name = 'l10nstats/compare.html'
+
+    def get_context_data(self, **kwargs):
+        context = super(CompareView, self).get_context_data(**kwargs)
+        try:
+            run = get_object_or_404(Run, id=self.request.GET.get('run'))
+        except ValueError:
+            raise Http404('Invalid ID')
+        doc = self.get_doc(run)
+        if doc and 'details' in doc:
+            if isinstance(doc['details'].get('children'), list):
```

Comment:
Would this be more explicit if it allowed the `KeyError` and used `JSON2Adaptor` in the `except` block? Or maybe this just needs a comment saying that there are two supported version of the JSON.